sortinghat/core/api.py

```diff
@@ -357,6 +358,68 @@ def move_identity(ctx, from_id, to_uuid):
     return uidentity
 
 
+@django.db.transaction.atomic
+def lock_unique_identity(ctx, uuid):
+    """Lock an unique identity so it cannot be modified
+
+    This function locks the unique identity identified by `uuid`
+    so this object and its related objects such as identities,
+    enrollments or the profile cannot be modified.
+
+    :param ctx: context from where this method is called
+    :param uuid: identifier of the unique identity which will be locked
+
+    :returns: a unique identity with its locked value updated
+
+    :raises InvalidValueError: raised when `uuid` is `None` or an empty string
+    """
+    if uuid is None:
+        raise InvalidValueError(msg="'uuid' cannot be None")
+    if uuid == '':
+        raise InvalidValueError(msg="'uuid' cannot be an empty string")
+
+    trxl = TransactionsLog.open('lock_unique_identity', ctx)
+
+    uidentity = find_unique_identity(uuid)
+
+    uidentity = set_lock_unique_identity_db(trxl, uidentity, True)
+
+    trxl.close()
```

Comment:
For readability, I think it would be clearer this 
```suggestion
    trxl = TransactionsLog.open('lock_unique_identity', ctx)
    
    uidentity = find_unique_identity(uuid)
    uidentity = set_lock_unique_identity_db(trxl, uidentity, True)
    
    trxl.close()
```

Reply:
Done!